File: tests/algorithms/x/Java/data_structures/binary_tree/mirror_binary_tree.java
public class Main {
    static class Tree {
        int[] values;
        int[] left;
        int[] right;
        int root;
        Tree(int[] values, int[] left, int[] right, int root) {
            this.values = values;
            this.left = left;
            this.right = right;
            this.root = root;
        }
        Tree() {}
        @Override public String toString() {
            return String.format("{'values': %s, 'left': %s, 'right': %s, 'root': %s}", String.valueOf(values), String.valueOf(left), String.valueOf(right), String.valueOf(root));
        }
    }


    static void mirror_node(int[] left, int[] right, int idx) {
        if (idx == (-1)) {
            return;
        }
        int temp = left[idx];
left[idx] = right[idx];
right[idx] = temp;
        mirror_node(((int[])(left)), ((int[])(right)), left[idx]);
        mirror_node(((int[])(left)), ((int[])(right)), right[idx]);
    }

    static Tree mirror(Tree tree) {
        mirror_node(((int[])(tree.left)), ((int[])(tree.right)), tree.root);
        return tree;
    }

    static int[] inorder(Tree tree, int idx) {
        if (idx == (-1)) {
            return new int[]{};
        }
        int[] left_vals = ((int[])(inorder(tree, tree.left[idx])));
        int[] right_vals = ((int[])(inorder(tree, tree.right[idx])));
        return concat(concat(left_vals, new int[]{tree.values[idx]}), right_vals);
    }

    static Tree make_tree_zero() {
        return new Tree(new int[]{0}, new int[]{-1}, new int[]{-1}, 0);
    }

    static Tree make_tree_seven() {
        return new Tree(new int[]{1, 2, 3, 4, 5, 6, 7}, new int[]{1, 3, 5, -1, -1, -1, -1}, new int[]{2, 4, 6, -1, -1, -1, -1}, 0);
    }

    static Tree make_tree_nine() {
        return new Tree(new int[]{1, 2, 3, 4, 5, 6, 7, 8, 9}, new int[]{1, 3, -1, 6, -1, -1, -1, -1, -1}, new int[]{2, 4, 5, 7, 8, -1, -1, -1, -1}, 0);
    }

    static void main() {
        String[] names = ((String[])(new String[]{"zero", "seven", "nine"}));
        Tree[] trees = ((Tree[])(new Tree[]{make_tree_zero(), make_tree_seven(), make_tree_nine()}));
        int i = 0;
        while (i < trees.length) {
            Tree tree = trees[i];
            System.out.println("      The " + names[i] + " tree: " + _p(inorder(tree, tree.root)));
            Tree mirrored = mirror(tree);
            System.out.println("Mirror of " + names[i] + " tree: " + _p(inorder(mirrored, mirrored.root)));
            i = i + 1;
        }
    }
    public static void main(String[] args) {
        main();
    }

    static <T> T[] concat(T[] a, T[] b) {
        T[] out = java.util.Arrays.copyOf(a, a.length + b.length);
        System.arraycopy(b, 0, out, a.length, b.length);
        return out;
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/x/PHP/data_structures/binary_tree/mirror_binary_tree.php
<?php
ini_set('memory_limit', '-1');
function _str($x) {
    if (is_array($x)) {
        $isList = array_keys($x) === range(0, count($x) - 1);
        if ($isList) {
            $parts = [];
            foreach ($x as $v) { $parts[] = _str($v); }
            return '[' . implode(' ', $parts) . ']';
        }
        $parts = [];
        foreach ($x as $k => $v) { $parts[] = _str($k) . ':' . _str($v); }
        return 'map[' . implode(' ', $parts) . ']';
    }
    if (is_bool($x)) return $x ? 'true' : 'false';
    if ($x === null) return 'null';
    return strval($x);
}
function mirror_node(&$left, &$right, $idx) {
  if ($idx == (-1)) {
  return;
}
  $temp = $left[$idx];
  $left[$idx] = $right[$idx];
  $right[$idx] = $temp;
  mirror_node($left, $right, $left[$idx]);
  mirror_node($left, $right, $right[$idx]);
}
function mirror(&$tree) {
  mirror_node($tree['left'], $tree['right'], $tree['root']);
  return $tree;
}
function inorder($tree, $idx) {
  if ($idx == (-1)) {
  return [];
}
  $left_vals = inorder($tree, $tree['left'][$idx]);
  $right_vals = inorder($tree, $tree['right'][$idx]);
  return array_merge(array_merge($left_vals, [$tree[$values][$idx]]), $right_vals);
}
function make_tree_zero() {
  return ['values' => [0], 'left' => [-1], 'right' => [-1], 'root' => 0];
}
function make_tree_seven() {
  return ['values' => [1, 2, 3, 4, 5, 6, 7], 'left' => [1, 3, 5, -1, -1, -1, -1], 'right' => [2, 4, 6, -1, -1, -1, -1], 'root' => 0];
}
function make_tree_nine() {
  return ['values' => [1, 2, 3, 4, 5, 6, 7, 8, 9], 'left' => [1, 3, -1, 6, -1, -1, -1, -1, -1], 'right' => [2, 4, 5, 7, 8, -1, -1, -1, -1], 'root' => 0];
}
function main() {
  $names = ['zero', 'seven', 'nine'];
  $trees = [make_tree_zero(), make_tree_seven(), make_tree_nine()];
  $i = 0;
  while ($i < count($trees)) {
  $tree = $trees[$i];
  echo rtrim('      The ' . $names[$i] . ' tree: ' . _str(inorder($tree, $tree['root']))), PHP_EOL;
  $mirrored = mirror($tree);
  echo rtrim('Mirror of ' . $names[$i] . ' tree: ' . _str(inorder($mirrored, $mirrored['root']))), PHP_EOL;
  $i = $i + 1;
};
}
main();
